src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,97 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_SIZE = 10;
+    private static final String OF_ABSENT_MESSAGE = "Element of ArrayList ain't present";
+    private T[] storage;
+    private int size;
+
+    public ArrayList() {
+        this.storage = (T[]) new Object[START_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (storage.length == size) {
+            increaseSize();
+        }
+        storage[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        inspectIndex(index, size + 1);
+        if (storage.length < size + 1) {
```

Comment:
This will be much more readable
```suggestion
        if (storage.length == size) {
```

Reply:
i did change condition definition